app/controllers/api/v3/bikes.rb

```diff
@@ -0,0 +1,300 @@
+            - Can be viewed on the HTML site /bikes/{id} (same as non-test bikes)
+
+            *`test` is automatically marked true on this documentation page. Set it to false it if you want to create actual bikes*
+
+            **Ownership**: Bikes you create will be created by the user token you authenticate with, but they will be sent to the email address you specify.
+
+          NOTE
+        }
+        params do
+          requires :serial, type: String, desc: "The serial number for the bike"
+          requires :manufacturer, type: String, desc: "Manufacturer name or ID"
+          # [Manufacturer name or ID](api_v2#!/manufacturers/GET_version_manufacturers_format)
+          requires :owner_email, type: String, desc: "Owner email"
+          requires :color, type: String, desc: "Main color or paint - does not have to be one of the accepted colors"
+          optional :test, type: Boolean, desc: "Is this a test bike?"
+          optional :organization_slug, type: String, desc: "Organization bike should be created by. **Only works** if user is a member of the organization"
+          optional :cycle_type_name, type: String, values: CYCLE_TYPE_NAMES, default: 'bike', desc: "Type of cycle (case sensitive match)"
+          use :bike_attrs
+          optional :components, type: Array do
+            use :components_attrs
+          end
+        end
+        post '/', serializer: BikeV2ShowSerializer, root: 'bike' do
+          declared_p = { "declared_params" => declared(params, include_missing: false).merge(creation_state_params) }
+          # TODO: Changing this origin breaks things, but also the origin is incorrect. need to fix.
+          b_param = BParam.create(creator_id: creation_user_id, params: declared_p['declared_params'], origin: 'api_v2')
+          ensure_required_stolen_attrs(b_param.params)
+
+
+          # Prevents unnecessary DB hits
+          if b_param.errors.present? or b_param.bike_errors.present?
+            e = b_param.errors
+            return error!(e.full_messages.to_sentence, 401)
+          end
+
+          existing_bike = Bike.find_by(serial_normalized: SerialNormalizer.new({serial: b_param.serial_number}).normalized)
+
+          # Assume only one bike for now, but may need to check len of existing bike later depending on what Seth says
+          if existing_bike.present?
+            @bike = existing_bike
+            # Check organization ownership
+            authorize_bike_for_user
+            
+            # Additionally, check to ensure the email is attached to the bike (either as primary or secondary)
+            email_is_attached_to_bike_owner(b_param['params']['bike']['owner_email'])
+
+            begin
+              # We only want to update the Bike Attrs, which is a subset of the creation parameters
+              BikeUpdator.new(user: current_user, bike: existing_bike, b_params: BParam.v2_params(declared_p['declared_params'].slice(:bike_attrs).as_json)).update_available_attributes
```

Comment:
One interesting thing is that things like color and manufacturer get stripped here, but they're still marked as required by the endpoint. Is that okay?